src/dataframe/dataframe.jl

```diff
@@ -147,7 +152,7 @@ function DataFrame(d::AbstractDict)
     end
     colindex = Index([Symbol(k) for k in colnames])
     columns = Any[d[c] for c in colnames]
-    DataFrame(columns, colindex)
+    DataFrame(columns, colindex, copycolumns=copycolumns)
 end
 
 function DataFrame(; kwargs...)
```

Comment:
I recommended to disallow `copycolumns` kwarg in this constructor, but maybe we should allow it?

Reply:
Yes, I don't see why we shouldn't support it.

Reply:
It was just weird for me that e.g. `DataFrame(copycolumns=true)` creates an empty `DataFrame` and not a `DataFrame` with a single column and single row containing `true`.
But I can change it as probably `copycolumns` name for a column is not very likely. Just please confirm that you have not changed your opinion.

Reply:
Ah, right, that's the ambiguity problem. Though it sounds quite unlikely that one would end up passing `copycolumns=true/false` when intending to create a column. Can we just ensure that things like `copycolumns=1` and `copycolumns=[1]` do create a column?

Reply:
This is not something I would want to do. We should:
* either make `copycolumns` be `Bool` and never create a column*
* or it should always create a column and not be treated as an indicator if we want to copy columns.

Which one of these two do you prefer?

My point was that this constructor is probably mostly used in interactive work and not in production code (so I opted for the latter option, but I do not have a very strong preference here).

Reply:
OK, let's just assume people won't need to create a `copycolumns` or `copycols` column using this constructor. Indeed it's mainly intended for interactive use.

Reply:
OK - so I add special handling of `copycolumns` kwarg and add it to the documentation.